src/all.test.js

```diff
@@ -0,0 +1,91 @@
+/* eslint-env jest */
+/* global spyOn */
+import HMRCFrontend from './all';
+import * as characterCountModule from './components/character-count/character-count';
+
+describe('Init All', () => {
+  let testScope;
+  beforeEach(() => {
+    testScope = {
+      elemsToWipe: [],
+    };
+  });
+  afterEach(() => {
+    jest.resetAllMocks();
+    testScope.elemsToWipe.forEach((elem, key) => {
+      const parent = elem.parentNode;
+      if (parent) {
+        parent.removeChild(elem);
+      } else {
+        console.warn('failed to remove child', key);// eslint-disable-line no-console
+      }
+    });
+    testScope.elemsToWipe = [];
+  });
+  describe('Character Count', () => {
+    function createCharCountModule() {
+      const elem = document.createElement('div');
+      testScope.elemsToWipe.push(elem);
+      elem.setAttribute('data-module', 'hmrc-character-count');
+      document.querySelector('body').appendChild(elem);
+      return elem;
+    }
+
+    it('should initialise one module', () => {
+      const initFn = jest.fn();
+      const elem = createCharCountModule();
+      spyOn(characterCountModule, 'default').and.returnValue({ init: initFn });
+      expect(characterCountModule.default).not.toHaveBeenCalled();
+      HMRCFrontend.initAll();
+      expect(characterCountModule.default).toHaveBeenCalledWith(elem);
+      expect(initFn).toHaveBeenCalled();
+    });
+
+    it('should not initialise when there are no modules', () => {
+      const initFn = jest.fn();
+      spyOn(characterCountModule, 'default').and.returnValue({ init: initFn });
+      HMRCFrontend.initAll();
+      expect(characterCountModule.default).not.toHaveBeenCalled();
+      expect(initFn).not.toHaveBeenCalled();
+    });
+
+    it('should initialise four module', () => {
```

Comment:
Being really picky here but should be 'should initialise four modules'

Reply:
Thank you.

Reply:
@timsb I've pushed a new version with these changes.